src/renderer/components/dock/logs/controls.tsx

```diff
@@ -44,13 +47,26 @@ const NonInjectedLogControls = observer(({ openSaveFileDialog, model }: Dependen
     model.reloadLogs();
   };
 
+  const downloadAllLogs = async () => {
+    const pod = model.pod.get();
+    
+    if (pod) {
+      const logs = await callForAllLogs(pod, { timestamps: showTimestamps, previous });
+      
+      openSaveFileDialog(`${pod.getName()}.log`, logs, "text/plain");
+    }
+  };
+
   const downloadLogs = () => {
-    const fileName = pod.getName();
-    const logsToDownload: string[] = showTimestamps
-      ? model.logs.get()
-      : model.logsWithoutTimestamps.get();
+    return new Promise((resolve) => {
+      const fileName = pod.getName();
+      const logsToDownload: string[] = showTimestamps
+        ? model.logs.get()
+        : model.logsWithoutTimestamps.get();
 
-    openSaveFileDialog(`${fileName}.log`, logsToDownload.join("\n"), "text/plain");
+      openSaveFileDialog(`${fileName}.log`, logsToDownload.join("\n"), "text/plain");
+      resolve(true);
+    });
```

Comment:
Why does this need to be async?

Reply:
I agree, it doesn't look async at all. `openSaveFileDialog` in the other hand does more than name says. Maybe it could be renamed to `askDirectoryAndSaveToFile` and made async.

Reply:
Moved `downloadLogs` part to logs model and removed a Promise "wrapper".